airflow/models/taskinstance.py

```diff
@@ -190,23 +192,24 @@ def set_current_context(context: Context) -> Generator[Context, None, None]:
             )
 
 
-def _stop_remaining_tasks(*, self, session: Session):
+def _stop_remaining_tasks(*, task_instance: TaskInstance | TaskInstancePydantic, session: Session):
     """
     Stop non-teardown tasks in dag.
 
     :meta private:
     """
-    tis = self.dag_run.get_task_instances(session=session)
+    assert task_instance.dag_run is not None
```

Comment:
We only use asserts in in special circumstances - I suggest to change it to 

```
if not task_instance_.dag_run:
   raise ValueError("task_instance must have dag_run set")

```

or similar.